tools/pydfu.py: Refactor syntax, update comments and docstrings

c0rejump opened the pull request:
I'm learning :)

c0rejump committed:
tools/pydfu.py: Refactor syntax, update comments and docstrings

Some parts of code have been aligned to increase readability.
In general '' instead of "" were used wherever possible to keep
the same convention for entire file. Import inspect line has been
moved to the top according to hints reported by pep8 tools. Few
extra spaces were removed, few missing spaces were added. Comments
have been updated, mostly in "read_dfu_file" function. Some other
comments have been capitalized and/or slightly updated. Few
docstrings were fixed as well.

No real code changes intended.

dpgeorge commented:
Thanks for the contribution.  It looks OK to me, the only thing I'd say is to relax the hard line-length limit.

dpgeorge commented on tools/pydfu.py:
```diff
@@ -119,16 +127,17 @@ def init():
-        elif (status == __DFU_STATE_DFU_DOWNLOAD_IDLE
-            or status == __DFU_STATE_DFU_UPLOAD_IDLE):
+        elif (status == __DFU_STATE_DFU_DOWNLOAD_IDLE or
+              status == __DFU_STATE_DFU_UPLOAD_IDLE):
```
I think it's clearer with the "or" on the second line.

dpgeorge commented on tools/pydfu.py:
```diff
@@ -119,16 +127,17 @@ def init():
     """Sends an abort request."""
-    __dev.ctrl_transfer(0x21, __DFU_ABORT, 0, __DFU_INTERFACE, None, __TIMEOUT)
+    __dev.ctrl_transfer(0x21, __DFU_ABORT, 0, __DFU_INTERFACE,
+                        None, __TIMEOUT)
```
This doesn't need to be split, we usually go for a maximum line length of "about 100".

c0rejump committed:
tools/pydfu.py: Minor syntax updates

The "or" keyword moved inside "init" function (less confusing now).
Merged lines in "abort_request" function.

No real code changes intended.

c0rejump commented:
Hi,

Thanks for all comments. PR updated.

dpgeorge commented:
Thanks!  Rebased, squashed and merged in 6db5cede06dfb580c5e4f6cdeb1796256969b4fb
